packages/aws-cdk/lib/archive.ts

```diff
@@ -1,26 +1,39 @@
 import archiver = require('archiver');
 import crypto = require('crypto');
 import fs = require('fs-extra');
+import glob = require('glob');
+import path = require('path');
 
 export function zipDirectory(directory: string, outputFile: string): Promise<void> {
   return new Promise((ok, fail) => {
-    const output = fs.createWriteStream(outputFile);
-    const archive = archiver('zip');
     // The below options are needed to support following symlinks when building zip files:
-    // -  nodir: This will prevent symlinks themselves from being copied into the zip.
+    // - nodir: This will prevent symlinks themselves from being copied into the zip.
     // - follow: This will follow symlinks and copy the files within.
     const globOptions = {
       dot: true,
       nodir: true,
       follow: true,
-      cwd: directory
+      cwd: directory,
     };
-    archive.glob('**', globOptions);
-    archive.pipe(output);
-    archive.finalize();
+    const files = glob.sync('**', globOptions); // The output here is already sorted
+
+    const output = fs.createWriteStream(outputFile);
 
+    const archive = archiver('zip');
     archive.on('warning', fail);
     archive.on('error', fail);
+    archive.pipe(output);
+
+    files.forEach(file => { // Append files serially to ensure file order
+      const stream = fs.createReadStream(path.join(directory, file));
+      archive.append(stream, {
+        name: file,
+        date: new Date('1980-01-01T00:00:00.000Z'), // reset dates to get the same hash for the same content
```

Comment:
on my system `new Date(0)` seem to be translated to `1980-01-01T00:00:00.000Z` for the files in the zip... this means that I cannot write a unit test for the date reset => setting to `1980-01-01T00:00:00.000Z`